marxs/source/labSource.py

```diff
@@ -105,3 +108,74 @@ def generate_photons(self, exposuretime):
         photons.add_column(Column(name='dir', data=dir.T))
         photons.add_column(Column(name='polarization', data=polarization_vectors(dir.T, photons['polangle'])))
         return photons
+
+class LabPointSourceCone(Source):
+    '''In-lab point source
+
+    - photons uniformly distributed in all directions in cone
+    - photon start position is source position
+
+    Parameters
+    ----------
+    position: 3 element list
+        3D coordinates of photon source
+    direction: array
+        direciton is given by a vector [x,y,z]
+        This is the direction of the axis along which the beam travels.
+        It is sufficient to enter any vector that spans the beams axis.
+
+    delta: float 
+        This is half the openning angle of the cone
+
+    kwargs : see `Source`
+        Other keyword arguments include ``flux``, ``energy`` and ``polarization``.
+        See `Source` for details.
+    '''
+    def __init__(self, position, delta, direction=None,  **kwargs):
+
+        self.dir = direction / np.sqrt(np.dot(direction, direction)) #normalize direction
```

Comment:
This works. Becaue we need this so often, there also is a function for it, see
`transforms3d.unitils.normalize_vector`.